drop-transfer/src/manager.rs

```diff
@@ -138,7 +138,17 @@ impl TransferManager {
                 Ok(false)
             }
             Entry::Vacant(vacc) => {
-                self.storage.insert_transfer(&xfer.storage_info()).await;
+                if self
+                    .storage
+                    .insert_transfer(&xfer.storage_info())
+                    .await
+                    .is_none()
+                {
+                    warn!(self.logger, "Transfer was closed already");
```

Comment:
To be honest it was not closed, it was just there.. Closed would assume we have a Canceled state in DB

Reply:
Re-reading this code gives me more questions:
- We receive a transfer request
- We try to insert it
- If there was no such transfer, we insert it and check again. This looks like a bit unnecessary

- The message says the transfer was closed already but we have `cancel` in our semantics, so we should really check the state of the transfer if it contains a `cancel` state or else it just might be there not properly canceled as well

Reply:
Ok, I misused `closed` here. I meant `canceled`.
I don't agree with you. The source of truth is the hash map persisted in the form of sync tables. If there is no transfer in the hash map, it's not alive anymore (or does not exist), which is checked in this code.
Our logic does not make use of canceled state tables, these are for the clients of JSON API

Reply:
Thanks for explaining. It makes sense now. I was confused for a short moment.